src/services/collection.service.js

```diff
@@ -10,19 +12,47 @@ async function getCollectionByIdAndUserId(id, userId) {
   return user;
 }
 
+async function getProduct(pid) {
+  const products = await Products.findOne({
+    where: { id: pid },
+  });
+  if (!products) {
+    return [];
+  }
+  const images = await products.getProductImages();
+  const img = [];
+  for (let i = 0; i < images.length; i += 1) {
+    img.push(images[i].url);
+  }
+  const output = {
+    images: img,
+    product: products,
+  };
+  return output;
+}
+
```

Comment:
why are you editing the images, you send all images as object, and we can user eager loading to load the image instead of using to seperate queries